src/studySetup.js

```diff
@@ -0,0 +1,86 @@
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "getStudySetup" }]*/
+/* global VARIATIONS */
+
+/**
+ *  Overview:
+ *
+ *  - constructs a well-formatted `studySetup` for use by `browser.study.setup`
+ *  - mostly declarative, except that some fields are set at runtime
+ *    asynchronously.
+ *
+ *  Advanced features:
+ *  - testing overrides from preferences
+ *  - expiration time
+ *  - some user defined endings.
+ *  - study defined 'shouldAllowEnroll' logic.
+ */
+
+/** Base for studySetup, as used by `browser.study.setup`.
+ *
+ * Will be augmented by 'getStudySetup'
+ */
+const baseStudySetup = {
+  // used for activeExperiments tagging (telemetryEnvironment.setActiveExperiment)
+  activeExperimentName: browser.runtime.id,
+
+  // uses shield sampling and telemetry semantics.  Future: will support "pioneer"
+  studyType: "shield",
+
+  // telemetry
+  telemetry: {
+    // default false. Actually send pings.
+    send: true,
+    // Marks pings with testing=true.  Set flag to `true` before final release
+    removeTestingFlag: false,
+  },
+
+  // endings with urls
+  endings: {
+    /** standard endings */
```

Comment:
I think the FastBlock study just removed all of these endings, can we do that here?

Reply:
yep